src/layers/core/geojson-layer/geojson-layer.js

```diff
@@ -95,91 +101,106 @@ export default class GeoJsonLayer extends CompositeLayer {
         data: polygonFeatures,
         extruded,
         wireframe: true,
+        fp64,
         getPolygon: getCoordinates,
         getElevation,
-        getColor,
+        getColor: getLineColor,
         updateTriggers: {
           getElevation: updateTriggers.getElevation,
-          getColor: updateTriggers.getColor
+          getColor: updateTriggers.getLineColor
         },
         onHover,
         onClick
-      }));
+      });
 
-    const polygonOutlineLayer = !extruded &&
+    const polygonLineLayer = !extruded &&
       stroked &&
-      hasPolygonOutline &&
-      new PathLayer(Object.assign({}, this.props, {
+      hasPolygonLine &&
+      new PathLayer({
         id: `${id}-polygon-outline`,
         data: polygonOutlineFeatures,
+        widthScale: lineWidthScale,
+        widthMinPixels: lineWidthMinPixels,
+        widthMaxPixels: lineWidthMaxPixels,
+        rounded: lineJointRounded,
+        miterLimit: lineMiterLimit,
+        fp64,
         getPath: getCoordinates,
-        getColor,
-        getWidth,
-        updateTriggers: {
-          getColor: updateTriggers.getColor,
-          getWidth: updateTriggers.getWidth
-        },
+        getColor: getLineColor,
+        getWidth: getLineWidth,
         onHover,
-        onClick
-      }));
+        onClick,
+        updateTriggers: {
+          getColor: updateTriggers.getLineColor,
+          getWidth: updateTriggers.getLineWidth
+        }
+      });
 
-    const lineLayer = drawLines && new PathLayer(Object.assign({}, this.props, {
+    const pathLayer = drawLines && new PathLayer({
       id: `${id}-line-paths`,
       data: lineFeatures,
+      widthScale: lineWidthScale,
+      widthMinPixels: lineWidthMinPixels,
+      widthMaxPixels: lineWidthMaxPixels,
+      rounded: lineJointRounded,
+      miterLimit: lineMiterLimit,
+      fp64,
       getPath: getCoordinates,
-      getColor,
-      getWidth,
+      getColor: getLineColor,
+      getWidth: getLineWidth,
       onHover,
       onClick,
       updateTriggers: {
-        getColor: updateTriggers.getColor,
-        getWidth: updateTriggers.getWidth
+        getColor: updateTriggers.getLineColor,
+        getWidth: updateTriggers.getLineWidth
       }
-    }));
+    });
 
-    const pointLayer = drawPoints && new ScatterplotLayer(Object.assign({}, this.props, {
+    const pointLayer = drawPoints && new ScatterplotLayer({
       id: `${id}-points`,
       data: pointFeatures,
+      fp64,
       getPosition: getCoordinates,
       getColor: getFillColor,
       getRadius,
+      onHover,
+      onClick,
       updateTriggers: {
         getColor: updateTriggers.getFillColor,
         getRadius: updateTriggers.getRadius
-      },
-      onHover,
-      onClick
-    }));
+      }
+    });
 
     return [
       polygonFillLayer,
       polygonWireframeLayer,
-      polygonOutlineLayer,
-      lineLayer,
+      polygonLineLayer,
+      pathLayer,
       pointLayer
     ].filter(Boolean);
```

Comment:
We should to add the filtering by default inside deck.gl - Because I see so many layers doing `.filter(Boolean)` thing now. React supports `null`/empty elements, we should too. I think it is a one line change, could even go into this PR.